src/dsp_tools/commands/xmlupload/models/deserialise/xmlresource.py

```diff
@@ -112,10 +128,17 @@ def _get_properties(
                     ungrouped_properties.append(XMLProperty.from_node(subnode, "interval", default_ontology))
                 case "hasTitle" | "hasComment" | "hasDescription" | "hasKeyword":
                     ungrouped_properties.append(XMLProperty.from_node(subnode, "text", default_ontology))
-                case _:
+                case str() as x if x.endswith("-prop") and x.startswith(COMPOSITE_PROPS):
                     # get the property type which is in format type-prop, p.ex. <decimal-prop>
                     prop_type, _ = subnode.tag.split("-")
                     ungrouped_properties.append(XMLProperty.from_node(subnode, prop_type, default_ontology))
+                case _:
+                    raise XmlUploadError(f"Unexpected tag '{subnode.tag}'")
+        grouped_properties = XMLResource._group_props(ungrouped_properties)
```

Comment:
I would prefer this to be a top level function rather than a static method, as it doesn't really have a direct connection to the class containing it. Also, private static methods aren't particularly meaningful as a concept, as "private" would indicate that it is only intended in the scope of the class, which means that you normally have an instance of the class at hand anyway, which renders some of the use cases for static-ness obsolete.  
And in Python it's entirely idiomatic to have top level functions - which renders practically all use cases for static methods obsolete.  
I would really only (if at all) use static methods when it has a very direct relation to the Class it lives on, like e.g. a factory method.

Reply:
Good considerations. However, I'm unsure if it applies here. This private static method is a helper of a helper of the (static) factory. I made all helpers static as well, but private.

Reply:
I'm aware of that, but I would advocate making helpers top-level functions - simply because I don't see a benefit in calling a function like that _on a class_ when you could just as well simply _call the function_.  
But in the end, this is cosmetic and doesn't make a real difference, so I will leave it up to you.